src/openvic-simulation/map/ProvinceInstance.cpp

```diff
@@ -185,6 +190,120 @@ void ProvinceInstance::_update_pops(DefineManager const& define_manager) {
+			modifier_sum.add_modifier_nullcheck(static_modifier_cache.get_non_coastal(), province_source);
+		}
+
+		// TODO - overseas, blockaded, no_adjacent_controlled, has_siege, occupied, nationalism, infrastructure
+	}
+
+	for (BuildingInstance const& building : buildings.get_items()) {
+		modifier_sum.add_modifier(building.get_building_type(), province_source);
+	}
+
+	modifier_sum.add_modifier_nullcheck(crime, province_source);
+
+	modifier_sum.add_modifier_nullcheck(province_definition.get_continent(), province_source);
+
+	modifier_sum.add_modifier_nullcheck(province_definition.get_climate(), province_source);
+
+	modifier_sum.add_modifier_nullcheck(terrain_type, province_source);
+}
+
+void ProvinceInstance::contribute_country_modifier_sum(ModifierSum const& owner_modifier_sum) {
+	modifier_sum.add_modifier_sum_exclude_source(owner_modifier_sum, this);
+}
+
+fixed_point_t ProvinceInstance::get_modifier_effect_value(ModifierEffect const& effect) const {
+	if constexpr (ADD_OWNER_CONTRIBUTION) {
+		return modifier_sum.get_effect(effect);
+	} else {
+		using enum ModifierEffect::target_t;
+
+		if (owner != nullptr) {
+			if ((effect.get_targets() & PROVINCE) == NO_TARGETS) {
+				// Non-province targeted effects are already added to the country modifier sum
+				return owner->get_modifier_effect_value(effect);
+			} else {
+				// Province-targeted effects aren't passed to the country modifier sum
+				return owner->get_modifier_effect_value(effect) + modifier_sum.get_effect(effect);
+			}
+		} else {
+			return modifier_sum.get_effect(effect);
+		}
+	}
+}
+
+fixed_point_t ProvinceInstance::get_modifier_effect_value_nullcheck(ModifierEffect const* effect) const {
+	if (effect != nullptr) {
+		return get_modifier_effect_value(*effect);
+	} else {
+		return fixed_point_t::_0();
+	}
+}
+
+void ProvinceInstance::push_contributing_modifiers(
+	ModifierEffect const& effect, std::vector<ModifierSum::modifier_entry_t>& contributions
+) const {
+	if constexpr (ADD_OWNER_CONTRIBUTION) {
+		modifier_sum.push_contributing_modifiers(effect, contributions);
+	} else {
+		using enum ModifierEffect::target_t;
+
+		if (owner != nullptr) {
+			if ((effect.get_targets() & PROVINCE) == NO_TARGETS) {
+				// Non-province targeted effects are already added to the country modifier sum
+				owner->push_contributing_modifiers(effect, contributions);
+			} else {
+				// Province-targeted effects aren't passed to the country modifier sum
+				modifier_sum.push_contributing_modifiers(effect, contributions);
+				owner->push_contributing_modifiers(effect, contributions);
+			}
+		} else {
+			modifier_sum.push_contributing_modifiers(effect, contributions);
+		}
+	}
+}
+
+std::vector<ModifierSum::modifier_entry_t> ProvinceInstance::get_contributing_modifiers(ModifierEffect const& effect) const {
+	if constexpr (ADD_OWNER_CONTRIBUTION) {
+		return modifier_sum.get_contributing_modifiers(effect);
+	} else {
+		std::vector<ModifierSum::modifier_entry_t> contributions;
```

Comment:
Shouldn't you initialise it?

Reply:
All struct and class variables get initialised automatically (if they have a default constructor). It's only primitives that always need explicit initialisation (so basic integer and float types, enums, pointers, references, etc.), and even then that's mainly because of MSVC, I believe other compilers will auto initialise even primitive variables.